File: Trainer-backend/Trainer-backend/Model/Routine.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;

namespace Trainer_backend.Model
{
    [Serializable]
    public class Routine: BaseEntity
    {
        [DataMember]
        public virtual List<WorkSet> WorkSets { get; set; } = new ();
        [Required]
        [DataMember]
        public DateTime Date { get; set; }
    }
}

File: Trainer-frontend/src/app/module/routine.ts
import {Workset} from "./workset";

export class Routine {
  id: number;
  worksets: Workset[];
  date: Date;

  constructor(id: number = 0, worksets: Workset[], date: Date = new Date(Date.now())) {
    this.id = id;
    this.worksets = worksets ?? [];
    this.date = date;
  }

}
